jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextHandler.java

```diff
@@ -1198,14 +1198,11 @@ protected ServletContextRequest wrap(Request request)
             return null;
 
         // Get a servlet request, possibly from a cached version in the channel attributes.
-        // TODO We should cache this heavy weight object!  Something like:
-        // TODO: ServletChannel is not properly cleared out so I have disabled the caching of this for now.
-        ServletChannel servletChannel = null; // (ServletChannel)request.getComponents().getCache().get("blah.blah.ServletChannel");
+        ServletChannel servletChannel = (ServletChannel)request.getComponents().getCache().get(ServletChannel.class.getName());
```

Comment:
This wont work for HTTP2 as there can be concurrent requests for the same connection

Reply:
Oh yes it will, cache is scoped to `HttpChannelState`